src/wallet/crypter.cpp

```diff
@@ -356,140 +352,119 @@ bool CCryptoKeyStore::GetHDSeed(HDSeed& seedOut) const
 
 bool CCryptoKeyStore::AddKeyPubKey(const CKey& key, const CPubKey &pubkey)
 {
-    {
-        LOCK(cs_KeyStore);
-        if (!IsCrypted())
-            return CBasicKeyStore::AddKeyPubKey(key, pubkey);
+    LOCK(cs_KeyStore);
+    if (!fUseCrypto)
+        return CBasicKeyStore::AddKeyPubKey(key, pubkey);
 
-        if (IsLocked())
-            return false;
+    if (IsLocked())
+        return false;
 
-        std::vector<unsigned char> vchCryptedSecret;
-        CKeyingMaterial vchSecret(key.begin(), key.end());
-        if (!EncryptSecret(vMasterKey, vchSecret, pubkey.GetHash(), vchCryptedSecret))
-            return false;
+    std::vector<unsigned char> vchCryptedSecret;
+    CKeyingMaterial vchSecret(key.begin(), key.end());
+    if (!EncryptSecret(vMasterKey, vchSecret, pubkey.GetHash(), vchCryptedSecret))
+        return false;
 
-        if (!AddCryptedKey(pubkey, vchCryptedSecret))
-            return false;
-    }
-    return true;
+    return AddCryptedKey(pubkey, vchCryptedSecret);
```

Comment:
Just simplified the `return` logic slightly (here and in a few other places below), no functional change.